src/dsl.jl

```diff
@@ -110,158 +50,207 @@ end
 """
     @reaction_network
 
-Generates a [`ReactionSystem`](@ref dsl_description) that encodes a chemical reaction
-network.
+Macro for generating chemical reaction network models. Outputs a [`ReactionSystem`](@ref) structure,
```

Comment:
The old description was pretty much the first documentation of the package, and over 6 years old. Given the important to the package, updating this one seemed long overdue.